### PR TITLE
fix ip VIC plugin throws a parsing error when customer tries to speci…

### PR DESCRIPTION
**Pull request by zhoumeina**
issue description:
When you provide a gateway for the management network, it's mandatory to provide at least one routing destination. UI Plugin doesn't reflect that. It has two separate fields for Gateway and Routing destinations. If the customer doesn't provide both the fields properly, plugin throws a weird parsing error [Refer to the screenshot].

Fix:
1 Modify UI to make routing destination required.
2 Modify the components destination input and ip address to one input.
Signed-off-by: Meina Zhou <meinaz@vmware.com>

Fixes #

PR acceptance checklist:

[ ] All unit tests pass
[ ] All e2e tests pass
[ ] Unit test(s) included*
[ ] e2e test(s) included*
[ ] Screenshot attached and UX approved*

 *if applicable, add n/a if not

### CODE REVIEW

**Comment by zjs** [comment]
Adding @javierfz1980 and @lmalvins as I'm not a familiar enough with our UI code to properly review this.

File: h5c/vic/src/vic-webapp/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,13461 @@
+{
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/package-lock.json]: This seems unrelated to the change being made. Is this supposed to be included in this PR?

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.ts [hunk 2]
```diff
@@ -70,6 +70,7 @@ export class NetworksComponent implements OnInit {
         Validators.pattern(ipPattern)
       ]],
       clientNetworkRouting: [{ value: '', disabled: true }, [
+        Validators.required,
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.ts]: It looks like this may make the field required. I'm not sure that's the correct behavior. It seems like we want to require that either both `client-network-routing` and `client-network-gateway` are supplied or both are blank.

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.html [hunk 3]
```diff
@@ -296,73 +258,41 @@
...
-            <span class="tooltip-content" *ngIf="form.get('clientNetworkGateway').hasError('required')">
-              Client network gateway cannot be empty
+            <span class="tooltip-content" *ngIf="form.get('clientNetworkRouting').hasError('required')">
+              Routing destination cannot be empty
             </span>
-
-            <span class="tooltip-content" *ngIf="form.get('clientNetworkGateway').hasError('pattern')">
-              Gateway address is not valid
+            <span class="tooltip-content" *ngIf="form.get('clientNetworkRouting').hasError('pattern')">
+              IP addresses are not valid,please input one or more routing destinations in a comma separated list, e.g. 10.1.0.0/16,10.2.0.0/16:10.0.0.1
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.html]: > IP addresses are not valid,please input ...

There's a space missing after the comma ("valid, please").

> e.g. 10.1.0.0/16,10.2.0.0/16:10.0.0.1

This content seems to be for the routing destinations field. Maybe it should exclude the ":10.0.0.1" from the example, since that goes in the other field?

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.html [hunk 4]
```diff
@@ -296,73 +258,41 @@
...
-                   formControlName="clientNetworkRouting">
-
-            <span class="tooltip-content" *ngIf="form.get('clientNetworkRouting').hasError('pattern')">
-              IP addresses are not valid
+            <span class="tooltip-content" *ngIf="form.get('clientNetworkGateway').hasError('required')">
+              Client network gateway cannot be empty
             </span>
 
+            <span class="tooltip-content" *ngIf="form.get('clientNetworkGateway').hasError('pattern')">
+              Gateway address is not valid, please input ip address with format *.*.*.*. e.g. 10.1.119.23
```
Comment (zjs) [inline h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.html]: > ip address

It seems like we capitalize "IP" elsewhere in the UI.

> with format *.*.*.*. e.g. 10.1.119.23

This may just be personal preference, but I find the "*.*.*.*" hard to read. I think most users should understand how to format an IP address, so the example is probably sufficient.

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.html (hunk 3 above)
Comment (zhoumeina) [reply h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.html]: done.

File: h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.ts (hunk 2 above)
Comment (renmaosheng) [reply h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.ts]: Zach, for the client-network-gateway is it also set for the default route which vCH assigned IP range belongs to? or just for the optional IP range?

Comment (renmaosheng) [reply h5c/vic/src/vic-webapp/src/app/create-vch-wizard/networks/networks.component.ts]: screen shot 2018-08-31 at 9 48 30 am
I took a further look at the previous version, I don't think we even need this change. as this change performs the similar functionality as before.

File: h5c/vic/src/vic-webapp/package-lock.json (hunk 1 above)
Comment (zhoumeina) [reply h5c/vic/src/vic-webapp/package-lock.json]: yes, not related with this issue, but necessary

**Comment by zhoumeina** [comment]
@zjs , now I modify the destination routing to be optional. But if users input the static ip and gateway without destination routing, backend will return "parsing vch body from "" failed, because json: cannot unmarshal object into Go struct field Network.static of type models.CIDR" error message. I tried to not pass destination routing in the payload, also tried to set it empty. All returns same error. I believe logic should be modified in the backend to support it.
